pandas/types/missing.py

```diff
@@ -391,3 +394,30 @@ def na_value_for_dtype(dtype):
     elif is_bool_dtype(dtype):
         return False
     return np.nan
+
+
+def validate_fill_value(value, dtype):
+    """
+    Make sure the fill value is appropriate for the given dtype.
+    """
+    if not is_scalar(value):
+        raise TypeError('"fill_value" parameter must be '
+                        'a scalar, but you passed a '
+                        '"{0}"'.format(type(value).__name__))
+    elif not isnull(value):
```

Comment:
actually you already check that ``value is None`` is ok (just need a test to check!)